src/ripple/basics/base_uint.h

```diff
@@ -241,10 +241,15 @@ class base_uint
         return *this;
     }
 
-    // be32toh and htobe32 are macros that somehow cause shadowing
-    // warnings in this header file, so we hide them...
-    static uint32_t bigendToHost (uint32_t x) { return be32toh(x); }
-    static uint32_t hostToBigend (uint32_t x) { return htobe32(x); }
+    static uint32_t bigendToHost (uint32_t x)
+    {
+        return boost::endian::big_to_native(x);
+    }
+
+    static uint32_t hostToBigend (uint32_t x)
+    {
+        return boost::endian::native_to_big(x);
+    }
```

Comment:
These aren't used anywhere outside of `base_uint`.  How about removing these and just using `boost::endian` directly?